pkgs/servers/uwsgi/default.nix

```diff
@@ -83,6 +89,12 @@ stdenv.mkDerivation rec {
     substituteAll ${./nixos.ini} buildconf/nixos.ini
   '';
 
+  # this is a hack to make the php plugin link with session.so (which on nixos is a separate package)
+  # the hack works in coordination with ./0002-additional-php-ldflags.patch
+  UWSGICONFIG_PHP_LDFLAGS = if (builtins.any (x: x.name == "php") needed)
```

Comment:
Can you use `stdenv.lib.optionalString` here? Maybe also `concatStringsSep ","` to break up the line.